core/lib/da_client/src/types.rs

```diff
@@ -0,0 +1,43 @@
+use std::{error, fmt::Display};
+
+use serde::Serialize;
+
+/// `DAError` is the error type returned by the DA clients.
+#[derive(Debug)]
+pub struct DAError {
+    pub error: anyhow::Error,
+    pub is_transient: bool,
+}
+
+impl DAError {
+    pub fn is_transient(&self) -> bool {
+        self.is_transient
+    }
+}
+
+impl Display for DAError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "DAError: {}, is_transient: {}",
+            self.error, self.is_transient
+        )
```

Comment:
Nit: `Display` implementation (esp. for errors) is supposed to be more high-level / human-readable, something like: "transient / fatal data availability client error: {}".